packages/react-dom-bindings/src/client/ReactDOMInput.js

```diff
@@ -264,9 +260,9 @@ export function initInput(
     // Only assign the checked attribute if it is defined. This saves
     // a DOM write when controlling the checked attribute isn't needed
     // (text inputs, submit/reset)
-    if (props.defaultChecked != null) {
+    if (defaultChecked != null) {
       node.defaultChecked = !node.defaultChecked;
```

Comment:
(what does this extra assignment even do? seems like it should have no effect:
> The checked content attribute is a boolean attribute that gives the default checkedness of the input element. When the checked content attribute is added, if the control does not have dirty checkedness, the user agent must set the checkedness of the element to true; when the checked content attribute is removed, if the control does not have dirty checkedness, the user agent must set the checkedness of the element to false.


)

Reply:
Need investigation if it's a browser quirk. I could imagine browsers storing attribute and property values separately and missing a case where they should be updated and therefore not aligning. But not sure why this was added.